source/Ship.cpp

```diff
@@ -2413,10 +2414,20 @@ void Ship::Recharge(int rechargeType, bool hireCrew)
 	pilotError = 0;
 	pilotOkay = 0;
 
-	if((rechargeType & Port::RechargeType::Shields) || attributes.Get("shield generation"))
+	if(attributes.Get("shield generation"))
+		shields = max(shields, MaxShields() * ShieldRepairLimit());
+	if(attributes.Get("hull repair rate"))
+		hull = max(hull, MaxHull() * HullRepairLimit());
```

Comment:
this wont cause a sudden drop?
I mean if you got almost full hull, it tries to repair and then it drops to the limit?

Reply:
Makes sense:+1: